README.md

```diff
@@ -131,6 +131,47 @@ cpu, err := ghw.CPU(ghw.WithSnapshot(ghw.SnapshotOptions{
 }))
 ```
 
+### Creating snapshots
+
+You can create ghw snapshots in two ways.
+You can just consume the `ghw-snapshot` tool, or you can create them programmatically
+from your golang code. We explore now the latter case.
+
+Snapshotting takes two phases:
+1. clone the relevant pseudofiles/pseudodirectories into a temporary tree
+   This tree is usually deleted once the packing is successful.
+2. pack the cloned tree into a tar.gz
+
+```go
+scratchDir, err := ioutil.TempDir("", "ghw-snapshot-*")
+if err != nil {
+	fmt.Printf("Error creating clone directory: %v", err)
+}
+defer os.RemoveAll(scratchDir)
+
+// this step clones all the files and directories ghw cares about
+if err := snapshot.CloneTreeInto(scratchDir); err != nil {
+	fmt.Printf("error cloning into %q: %v", scratchDir, err)
+}
+
+// optionally, you may add extra content into your snapshot.
+// ghw will ignore the extra content.
+// Glob patterns like `filepath.Glob` are supported.
+fileSpecs := []string{
+	"/proc/cmdline",
+}
+// options allows the client code to optionally deference symlinks, or copy
+// them into the cloned tree as symlinks
```

Comment:
for the above comment to make sense, it probably would be good to do something like this:

```go
var opts *snapshot.CopyFileOptions
if err := snapshot.CopyFilesInto(fileSpecs, scratchDir, opts); err != nil {
```

Reply:
good point. Done.